docs/docs/dev-setup.md

```diff
@@ -4,9 +4,189 @@ title: Basic Setup
+
+### Debian/Ubuntu
+
+On Debian and Ubuntu, this can be accomplished with:
+
+```bash
+apt-get install -y \
+    git \
+    wget \
+    autoconf \
+	automake \
+	build-essential \
+	ccache \
+	device-tree-compiler \
+	dfu-util \
+	g++ \
+	gcc \
+	gcc-multilib \
+	libtool \
+	make \
+    cmake \
+	python3-dev \
+	python3-pip \
+	python3-setuptools \
+	xz-utils
+```
+
+### Fedora
+
+TODO
+
+### macOS
+
+TODO
+
+### WSL
+
+TODO
+
+## Setup
+
+### West Build Command
+
+`west` is the [Zephyr™ meta-tool](https://docs.zephyrproject.org/latest/guides/west/index.html) used to configure and build Zephyr™ applications. It can be installed by using the `pip` python package manager:
+
+```bash
+pip3 install --user west
+```
+
+:::note
+If you don't already have it configured, you may need to update your
+`PATH` to include the pip install path. See [User Installs](https://pip.pypa.io/en/stable/user_guide/#user-installs) and [Stack Overflow](https://stackoverflow.com/questions/38112756/how-do-i-access-packages-installed-by-pip-user) for more details.
+:::
+
+### Zephyr™ ARM SDK
+
+To build firmwares for the ARM architecture (all supported MCUs/keyboards at this point), you'll need to install the Zephyr™ ARM SDK to your system:
+
+```
+export ZSDK_VERSION=0.11.2
+wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZSDK_VERSION}/zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" && \
+	sh "zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" --quiet -- -d /opt/toolchains/zephyr-sdk-${ZSDK_VERSION} && \
+	rm "zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run"
+```
+
+The installation will prompt with several questions about installation location, and creating a default `~/.zephyrrc` for you with various variables. The defaults shouldn normally work as expected.
+
+### Source Code
+
+Next, you'll need to clone the ZMK source repository if you haven't already:
+
+```
+git clone https://github.com/zmkfirmware/zmk.git
+```
+
+### Initialize & Update Zephy Workspace
+
+Since ZMK is built as a Zephyr™ application, the next step is
+to use `west` to initialize and update your workspace. The ZMK
+Zephyr™ application is in the `app/` source directory:
+
+#### Initialize West
+
+```bash
+west init -l app/
+```
+
+#### Update To Fetch Modules
+
+```bash
+west update
+```
+
+#### Export Zephyr™ Core
+
+```bash
+west zephyr-export
+```
+
+### Environment Variables
+
+By default, the Zephyr™ SDK will create a file named `~/.zephyrrc` with the correct environment variables to build ZMK.
+We suggest two main [options](https://docs.zephyrproject.org/latest/guides/env_vars.html?highlight=zephyrrc) for how to load those settings.
+
+#### Per Shell
+
+To load the Zephyr environment properly for just one transient shell, run:
+
+```
+source zmk/zephyr/zephyr-env.sh
+```
+
+#### All Shells
+
+To load the environment variables for your shell every time,
+append the existing `~/.zephyrrc` file to your shell's RC file, e.g.
+
+##### Bash
+
+```
+cat ~/.zephyrc >> ~/.bashrc
```

Comment:
No such file or directory :(

Reply:
Ah it was located at `/root/.zephyrrc`... Also typo here, not `zephyrc`, `zephyrrc`

Reply:
Had to `source ~/.bashrc` also since the one time command didn't work.